openedx_ledger/api.py

```diff
@@ -63,19 +67,37 @@ def reverse_full_transaction(transaction, idempotency_key, **metadata):
         return reversal
 
 
-def create_ledger(unit, idempotency_key, **metadata):
+def create_ledger(unit=None, idempotency_key=None, subsidy_uuid=None, initial_deposit=None, **metadata):
     """
-    Idempotency key here, too.
+    Primary interface for creating a Ledger record.
+
+    params:
+      unit: Optional unit for the ledger, defaults to the model default of USD_CENTS.
+      idempotency_key: Optional idempotency key, defaults to result of
+        the utility function for creating ledger idempotency key
+      subsidy_uuid: Optional subsidy uuid used in above utility function.
+      initial_deposit: Optional amount of value to initialize the ledger with.  In units specified by units
+        (units by default are USD_CENTS).
+      metadata: Optional additional information for the ledger.
     """
     ledger, _ = models.Ledger.objects.get_or_create(
-        unit=unit,
-        idempotency_key=idempotency_key,
-        metadata=metadata,
+        unit=unit or models.UnitChoices.USD_CENTS,
+        idempotency_key=idempotency_key or utils.create_idempotency_key_for_ledger(subsidy_uuid),
+        defaults={
+            'metadata': metadata,
+        },
     )
-    return ledger
-
+    if initial_deposit:
+        initial_idpk = utils.create_idempotency_key_for_transaction(
+            ledger,
+            initial_deposit,
+            is_initial_deposit=True,
+        )
+        create_transaction(
+            ledger,
+            initial_deposit,
+            initial_idpk,
+            state=models.TransactionStateChoices.COMMITTED,
+        )
```

Comment:
You may want to take a look at a pattern that I've come up with on the subsidy side of things:

https://github.com/openedx/enterprise-subsidy/blob/f6500ac9b50a30319081aa7530b2b2f1eab42575/enterprise_subsidy/apps/subsidy/models.py#L191-L203

I'm open to either method.  One thing that would have to be done in order to shift the logic into this area of the code is a moderate amount of un-tangling the unit tests and test factories to use this different assumption about how the initial transaction gets created.

Reply:
Yeah I just started diving into that this afternoon.